### PR TITLE
Converted sealed data types to be data classes

### PR DESCRIPTION
**Pull request by shamsasari**

### CODE REVIEW

File: core/src/main/kotlin/net/corda/core/node/services/ServiceType.kt [hunk 1]
```diff
@@ -35,21 +32,22 @@ sealed class ServiceType(val id: String) {
...
 
-    override operator fun equals(other: Any?): Boolean = (other is ServiceType) && (other.id == this.id)
+    fun getSubType(subTypeId: String): ServiceType = baseWithSubType(id, subTypeId)
 
     fun isSubTypeOf(superType: ServiceType) = (id == superType.id) || id.startsWith(superType.id + ".")
     fun isNotary() = isSubTypeOf(notary)
     fun isValidatingNotary() = isNotary() && id.contains(".validating")
     fun isNetworkMap() = id == networkMap.id
 
+    override fun equals(other: Any?): Boolean = other === this || other is ServiceType && other.id == this.id
```
Comment (adagys) [inline core/src/main/kotlin/net/corda/core/node/services/ServiceType.kt]: could ServiceType be a data class?

File: finance/src/test/kotlin/net/corda/contracts/CommercialPaperTests.kt [hunk 2]
```diff
@@ -197,11 +195,6 @@ class CommercialPaperTestsGeneric {
         }
     }
 
-    fun cashOutputsToVault(vararg outputs: TransactionState<Cash.State>): Pair<LedgerTransaction, List<StateAndRef<Cash.State>>> {
```
Comment (adagys) [inline finance/src/test/kotlin/net/corda/contracts/CommercialPaperTests.kt]: why is this being removed?

File: node/src/main/kotlin/net/corda/node/services/statemachine/FlowSession.kt [hunk 3]
```diff
@@ -32,12 +32,12 @@ sealed class FlowSessionState {
     abstract val sendToParty: Party
 
     /** [otherParty] may be a specific peer or a service party */
-    class Initiating(val otherParty: Party) : FlowSessionState() {
+    data class Initiating(val otherParty: Party) : FlowSessionState() {
         override val sendToParty: Party get() = otherParty
         override fun toString(): String = "${javaClass.simpleName}($otherParty)"
```
Comment (adagys) [inline node/src/main/kotlin/net/corda/node/services/statemachine/FlowSession.kt]: I'm guessing we don't need these toString() anymore

File: node/src/main/kotlin/net/corda/node/services/statemachine/FlowSession.kt [hunk 4]
```diff
@@ -32,12 +32,12 @@ sealed class FlowSessionState {
...
     }
 
-    class Initiated(val peerParty: Party, val peerSessionId: Long) : FlowSessionState() {
+    data class Initiated(val peerParty: Party, val peerSessionId: Long) : FlowSessionState() {
         override val sendToParty: Party get() = peerParty
         override fun toString(): String = "${javaClass.simpleName}($peerParty, $peerSessionId)"
```
Comment (adagys) [inline node/src/main/kotlin/net/corda/node/services/statemachine/FlowSession.kt]: same here

File: client/jfx/src/main/kotlin/net/corda/client/jfx/model/TransactionDataModel.kt [hunk 5]
```diff
@@ -54,22 +58,13 @@ data class PartiallyResolvedTransaction(
     }
 }
 
-sealed class TransactionCreateStatus(val message: String?) {
```
Comment (shamsasari) [inline client/jfx/src/main/kotlin/net/corda/client/jfx/model/TransactionDataModel.kt]: Nope.

File: core/src/main/kotlin/net/corda/core/node/services/ServiceType.kt (hunk 1 above)
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/node/services/ServiceType.kt]: Making it into a data class means making the constructor public, which isn't what we want I think.

File: finance/src/test/kotlin/net/corda/contracts/CommercialPaperTests.kt (hunk 2 above)
Comment (shamsasari) [inline finance/src/test/kotlin/net/corda/contracts/CommercialPaperTests.kt]: I was changing the `General()` below to `General` but Intellij said this method isn't used, so...

**Comment by shamsasari** [comment]
Back to you.

**Pull request merged**